### PR TITLE
Add user agent to default headers

### LINKED ISSUE #80: Add Package Version as User-Agent header
Simple implementation would be a static value. Could potentially fetch this dynamically.

### PR DESCRIPTION
**Pull request by bh2smith**
This is an attempt to dynamically fetch the users package version of dune-client and pass along with requests as `User-Agent`.

We can't test this directly because this package does not depend on itself. Will have to try it out. Anyway, the fallback mechanism uses a hard-coded value (i.e. the next version).

Closes #80

### CODE REVIEW

File: dune_client/base_client.py [hunk 1]
```diff
@@ -44,4 +46,5 @@ def api_version(self) -> str:
 
     def default_headers(self) -> Dict[str, str]:
         """Return default headers containing Dune Api token"""
-        return {"x-dune-api-key": self.token}
+        client_version = get_package_version("dune-client") or "1.3.0"
+        return {"x-dune-api-key": self.token, "User-Agent": client_version}
```
[the remark this answers is not in the document]
Comment (bh2smith) [reply dune_client/base_client.py:50]:
```suggestion
        return {"x-dune-api-key": self.token, "User-Agent": f"dune-client==v{client_version}"}
```

how does this look?

[the remark this answers is not in the document]
Comment (RichardKeo) [reply dune_client/base_client.py:50]: I wonder if we could somehow make it more explicit that this is the python library.

Maybe `dune-client/{client_version} (https://pypi.org/project/dune-client/)`?

[the remark this answers is not in the document]
Comment (bh2smith) [reply dune_client/base_client.py:50]: It seems a bit verbose considering that my understanding of the purpose is for dune's internal logging/debugging. However it doesn't bother me to go with that.

[the remark this answers is not in the document]
Comment (bh2smith) [reply dune_client/base_client.py]: Done in c15d820

**Pull request merged**